src/Post/Filter/IdFilter.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Post\Filter;
+
+use Flarum\Filter\FilterInterface;
+use Flarum\Filter\FilterState;
+
+class IdFilter implements FilterInterface
+{
+    public function getFilterKey(): string
+    {
+        return 'id';
+    }
+
+    public function filter(FilterState $filterState, string $filterValue, bool $negate)
+    {
+        $idString = trim($filterValue, '"');
+        $ids = explode(',', $idString);
+
+        $filterState
+            ->getQuery()
+            ->whereIn('posts.id', $ids, 'and', $negate)
+            ->orderByRaw("FIELD(id, $idString)");
```

Comment:
I don't think this is necessary, and it will probably conflict with the order set by the request. Also we should avoid using raws, this for example will lead to `$idString` being evaluated in the sql query (sql injection), so I can do something like `?filter[id]=66,0),(created_at` which will also change the sql code to order by creation date.

Reply:
Ah yes, the first law of web-based programming: don't insert user-provided content into raw DB queries. How did I not think of this... And you're right: we now sort the returned posts in the frontend anyways.